src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -45,7 +46,15 @@ public function getArguments(Request $request, callable $controller, \Reflection
         $arguments = [];
 
         foreach ($this->argumentMetadataFactory->createArgumentMetadata($controller, $reflector) as $metadata) {
+            if ($attributes = $metadata->getAttributesOfType(ValueResolver::class)) {
+                $resolverClass = reset($attributes)->class;
+            }
+
             foreach ($this->argumentValueResolvers as $resolver) {
+                if ($resolver::class !== ($resolverClass ?? $resolver::class)) {
```

Comment:
Are we sure that there cannot be multiple value resolvers using the same class? Not sure it's a good practice but we should take it into account. Maybe we can make sure they are keyed by service id in the iterator and use that "name" instead?

Reply:
> Are we sure that there cannot be multiple value resolvers using the same class?

Indeed, did not think of this.

> Maybe we can make sure they are keyed by service id in the iterator and use that "name" instead?

Would that mean `ArgumentResolver::getDefaultArgumentValueResolvers` should return the same keys set by the FrameworkBundle?

Reply:
Ah yes probably indeed. That would be a small behavior change though, maybe it'd be enough to re-index the array implicitly (just before assigning it to `$this->argumentResolvers` in `ArgumentResolver::__construct()` (replacing integers by FQCNs while preserving the order)

Reply:
Gave it a try!

Reply:
Gave it a try and I think what bothers me the most is that the index is either a service id or a FQCN. WDYT?